File: erp/erpv01/Models/Entities/Stoklar.cs
using System;
using System.Collections.Generic;

namespace erpv01.Models.Entities;

public partial class Stoklar
{
    public int Id { get; set; }

    public string Kod { get; set; } = null!;

    public string Ad { get; set; } = null!;

    public string? Ad2 { get; set; }

    public int? StokTuru { get; set; }

    public string? AnaBirim { get; set; }

    public string? Barkod { get; set; }

    public string? TedarikciKodu { get; set; }

    public decimal? SatisFiyati1 { get; set; }

    public decimal? AlisFiyati { get; set; }

    public decimal? KdvOrani { get; set; }

    public decimal? En { get; set; }

    public decimal? Boy { get; set; }

    public decimal? Yukseklik { get; set; }

    public decimal? Agirlik { get; set; }

    public decimal? Hacim { get; set; }

    public string? TeknikAciklama { get; set; }

    public string? GenelAciklama { get; set; }

    public string? EkAlan1 { get; set; }

    public string? EkAlan2 { get; set; }

    public string? EkAlan3 { get; set; }

    public string? EkAlan4 { get; set; }

    public string? EkAlan5 { get; set; }

    public string? EkAlan6 { get; set; }

    public string? EkAlan7 { get; set; }

    public string? EkAlan8 { get; set; }

    public string? EkAlan9 { get; set; }

    public string? EkAlan10 { get; set; }

    public bool AktifMi { get; set; }

    public DateTime OlusturmaTarihi { get; set; }

    public string? OlusturanKullanici { get; set; }

    public DateTime? GuncellemeTarihi { get; set; }

    public string? GuncelleyenKullanici { get; set; }
}

File: entegrasyon_uygulamasi/stoklar.py
from rapor_calistir import rapor_calistir, oku_json
import pyodbc
from datetime import datetime
import time

def run_table_sync():
    cfg = oku_json()
    cs = cfg["connectionstring"]
    parts = dict(p.split("=",1) for p in cs.split(";") if "=" in p)

    conn = pyodbc.connect(
        f"DRIVER={{ODBC Driver 17 for SQL Server}};"
        f"SERVER={parts['Server']};"
        f"DATABASE={parts['Database']};"
        f"UID={parts['Uid']};"
        f"PWD={parts['Pwd']};"
        f"TrustServerCertificate=yes"
    )
    cur = conn.cursor()

    code, msg, data = rapor_calistir("D_ENT_06")
    if code != "200":
        print("API HATA:", msg)
        conn.close()
        return

    keys = [str(x["_key"]) for x in data]

    cur.execute("SELECT ek_alan_1 FROM stoklar")
    mevcutlar = [str(r[0]) for r in cur.fetchall()]

    for d in data:
        ek = str(d["_key"])
        kod = d["stokkartkodu"]
        ad = d["aciklama"]
        aktif = 1 if d["durum"] == "A" else 0
        suan = datetime.now()
        anabirim = d["anabirim"]
        ek2 = str(d["birim_key"])
        ek3 = str(d["ozellikler"])
        ek4 = str(d["resim_url"])

        ozelkod1 = str(d["ozel_kod1_aciklama"])
        ozelkod2 = str(d["ozel_kod2_aciklama"])
        ozelkod3 = str(d["ozel_kod3_aciklama"])
        ozelkod4 = str(d["ozel_kod4_aciklama"])
        ozelkod5 = str(d["ozel_kod5_aciklama"])

        en = str(d["en"])
        boy = str(d["boy"])
        yukseklik = str(d["yukseklik"])
        agirlik = str(d["agirlik"])
        

        if ek in mevcutlar:
            cur.execute("""
                UPDATE stoklar SET 
                    kod=?,
                    ad=?,
                    aktif_mi=?,
                    guncelleme_tarihi=?,
                    ana_birim=?,
                    ek_alan_2=?,
                    ek_alan_3=?,
                    ek_alan_4=?,
                    guncelleyen_kullanici='ENTEGRASYON_01',
                    ozel_kod01=?,
                    ozel_kod02=?,
                    ozel_kod03=?,
                    ozel_kod04=?,
                    ozel_kod05=?,
                    en=?,
                    boy=?,
                    yukseklik=?,
                    agirlik=?
                WHERE ek_alan_1=?
            """, (
                kod, ad, aktif, suan, anabirim,
                ek2, ek3, ek4,
                ozelkod1, ozelkod2, ozelkod3, ozelkod4, ozelkod5,
                en,boy,yukseklik,agirlik,
                ek
            ))
        else:
            cur.execute("""
                INSERT INTO stoklar (
                    ek_alan_1, kod, ad, aktif_mi,
                    olusturma_tarihi, olusturan_kullanici,
                    ana_birim, ek_alan_2, ek_alan_3, ek_alan_4,
                    ozel_kod01, ozel_kod02, ozel_kod03, ozel_kod04, ozel_kod05, en, boy, yukseklik, agirlik
                ) VALUES (
                    ?, ?, ?, ?, ?, 'ENTEGRASYON_01',
                    ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?
                )
            """, (
                ek, kod, ad, aktif, suan,
                anabirim, ek2, ek3, ek4,
                ozelkod1, ozelkod2, ozelkod3, ozelkod4, ozelkod5, en,boy,yukseklik,agirlik
            ))

    for ek in mevcutlar:
        if ek not in keys:
            cur.execute("DELETE FROM stoklar WHERE ek_alan_1=?", ek)

    cur.execute("""
        UPDATE entegrasyon_tanimlari 
        SET deger_1=? 
        WHERE kod='stoklar'
    """, datetime.now())

    conn.commit()
    conn.close()
    print("Stoklar başarıyla güncellendi.")


if __name__ == "__main__":
    while True:
        run_table_sync()
        cfg = oku_json()
        dakika = int(cfg.get("periyot_dakika_stoklar", 30))
        print(f"{dakika} dakika sonra tekrar çalışacak...")
        time.sleep(dakika * 60)
